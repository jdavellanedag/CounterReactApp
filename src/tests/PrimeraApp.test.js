import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Test on <PrimeraApp />", () => {
  test("should show <PrimeraApp /> correctly", () => {
    const saludo = "Hola Soy Julian";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should show subtitle send by props", () => {
    const saludo = "Hola Soy Julian";
    const subtitle = "Mi primera app";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitle} />
    );
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitle);
  });

  /*   test("should show Hola Soy Julian", () => {
    const saludo = "Hola Soy Julian";
    const { getByText } = render(<PrimeraApp saludo="Hola Soy Julian" />);
    expect(getByText(saludo)).toBeInTheDocument();
  }); */
});
