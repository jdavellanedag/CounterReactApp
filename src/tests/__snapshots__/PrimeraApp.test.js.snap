// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <PrimeraApp /> should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola Soy Julian
  </h1>
  <p>
    Mi primera app
  </p>
</Fragment>
`;
